src/coreclr/hosts/inc/coreclrhost.h

```diff
@@ -46,5 +46,22 @@ CORECLR_HOSTING_API(coreclr_execute_assembly,
             unsigned int* exitCode);
 
 #undef CORECLR_HOSTING_API
-                      
+
+struct SequencePointInfo
+{
+    int lineNumber, ilOffset;
+    char16_t* fileName;
+};
+
+struct MethodDebugInfo
+{
+    SequencePointInfo* points;
+    int size;
+};
+
+typedef int (*GetInfoForMethodDelegate)(const char*, unsigned int, MethodDebugInfo& methodDebugInfo);
+extern GetInfoForMethodDelegate getInfoForMethodDelegate;
```

Comment:
Actually, it is kind of strange to create the delegate in the coreruncommon.cpp, which would mean that anyone implementing another hosting app would have to add it there too. It seems it would better be part of the coreclr_initialize, completely hidden inside coreclr.